chia/data_layer/data_layer.py

```diff
@@ -167,23 +169,37 @@ async def get_root_history(self, store_id: bytes32) -> List[SingletonRecord]:
         for record in records:
             if prev is None or record.root != prev.root:
                 root_history.append(record)
+                prev = record
         return root_history
 
     async def _validate_batch(
         self,
         tree_id: bytes32,
         to_check: List[SingletonRecord],
         min_generation: int,
-        max_generation: int,
     ) -> bool:
-        last_checked_hash: Optional[bytes32] = None
+        root: Optional[Root] = await self.data_store.get_tree_root(tree_id=tree_id)
+        assert root is not None
```

Comment:
```suggestion
        root = await self.data_store.get_tree_root(tree_id=tree_id)
```

Reply:
The `Optional` was introduced here by the extra unneeded type hint.

https://github.com/Chia-Network/chia-blockchain/blob/dad8d7a221cf029754e3cd7c6412f85419f55688/chia/data_layer/data_store.py#L335

Reply:
Looks like down below there is a reuse of this same variable name that also drives the `Optional`.

```
root = await self.data_store.get_last_tree_root_by_hash(
```

Perhaps these names should be more descriptive.  Or maybe it would be helpful if exceptional cases like requesting a thing we don't have resulted in exceptions.

Reply:
so should I rename the 2nd root differently, so it can take an optional, while the first one can keep the Root only type?

Reply:
Different names is probably a good idea if the two uses are referring to different logical things.  Also, `Optional` may just be a pain here.  Is getting `None` out of `DataStore.get_last_tree_root_by_hash()` really a 'normal' case?  Or is it exceptional?  The pragmatic question is 'do you want to write all the return value checking at every call site at every layer?'  Or can it just be allowed to propagate up via an exception.